client/src/components/Form/FormCardTool.vue

```diff
@@ -0,0 +1,253 @@
+import FormMessage from "./FormMessage";
+import ToolFooter from "components/Tool/ToolFooter";
+import ToolHelp from "components/Tool/ToolHelp";
+import Webhooks from "mvc/webhooks";
+
+export default {
+    components: {
+        FormMessage,
+        ToolFooter,
+        ToolHelp,
+    },
+    props: {
+        id: {
+            type: String,
+            required: true,
+        },
+        version: {
+            type: String,
+            required: false,
+        },
+        title: {
+            type: String,
+            required: true,
+        },
+        description: {
+            type: String,
+            required: false,
+        },
+        sustainVersion: {
+            type: Boolean,
+            default: false,
+        },
+        options: {
+            type: Object,
+            required: true,
+        },
+        messageText: {
+            type: String,
+            required: true,
+        },
+        messageVariant: {
+            type: String,
+            default: "info",
+        },
+    },
+    data() {
+        return {
+            isFavorite: this.getFavorite(),
+            webhookDetails: [],
+        };
+    },
+    computed: {
+        versions() {
+            return this.options.versions;
+        },
+        showAddFavorite() {
+            return !this.getUser().isAnonymous() && !this.isFavorite;
+        },
+        showVersions() {
+            return !this.sustainVersion && this.versions && this.versions.length > 1;
+        },
+        availableVersions() {
+            const versions = this.versions.slice();
+            const index = versions.indexOf(this.version);
+            if (index > -1) {
+                versions.splice(index, 1);
+            }
+            return versions.reverse();
+        },
+        showDownload() {
+            const user = this.getUser();
+            return user && user.get("is_admin");
+        },
+        showLink() {
+            return this.options.sharable_url;
+        },
+    },
+    created() {
+        // add tool menu webhooks
+        Webhooks.load({
+            type: "tool-menu",
+            callback: function (webhooks) {
+                webhooks.each((model) => {
+                    const webhook = model.toJSON();
+                    if (webhook.activate && webhook.config.function) {
+                        this.webhookDetails.push({
+                            icon: `fa ${webhook.config.icon}`,
+                            title: webhook.config.title,
+                            onclick: function () {
+                                const func = new Function("options", webhook.config.function);
+                                func(this.options);
+                            },
+                        });
+                    }
+                });
+            },
+        });
+    },
+    methods: {
+        getUser() {
+            const galaxy = getGalaxyInstance();
+            return galaxy.user;
```

Comment:
Can you use the `CurrentUser` provider here ? I think that's the new pattern we want to establish.

Reply:
Using currentUser now. Thx.